app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -689,6 +678,8 @@ class HomeFragment : Fragment() {
     private fun subscribeToTopSites(): Observer<List<TopSite>> {
         return Observer<List<TopSite>> { topSites ->
             requireComponents.core.topSiteStorage.cachedTopSites = topSites
+            context?.settings()?.preferences?.edit()
```

Comment:
Should we just use the `topSitesSize` `intPreference` that you added in `Settings.kt` here?